src/main/java/org/elasticsearch/indices/recovery/RecoveryState.java

```diff
@@ -366,31 +404,43 @@ public long stopTime() {
             return stopTime;
         }
 
-        public void stopTime(long stopTime) {
-            this.time = Math.max(0, stopTime - startTime);
-            this.stopTime = stopTime;
+        public void stop() {
+            assert stopTime == 0 : "already stopped";
+            stopTime = Math.max(System.currentTimeMillis(), startTime);
```

Comment:
I can only assert it's non-negative as some of the steps can take 0ms if they do nothing. Will add.